storage/storage_dest.go

```diff
@@ -555,20 +593,22 @@ func (s *storageImageDestination) queueOrCommit(index int, info addedLayerInfo)
 	return nil
 }
 
-// getDiffIDOrTOCDigest returns the diffID for the specified digest or the digest for the TOC, if known.
-func (s *storageImageDestination) getDiffIDOrTOCDigest(uncompressedDigest digest.Digest) (digest.Digest, bool) {
+// getLayerID returns the diffID for the specified digest or the digest for the TOC, if known.
+func (s *storageImageDestination) getLayerID(uncompressedDigest digest.Digest, tocDigest digest.Digest) (digest.Digest, bool) {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
-	if d, found := s.diffOutputs[uncompressedDigest]; found {
-		return d.TOCDigest, found
+	if tocDigest != "" {
+		if d, found := s.tocDigests[uncompressedDigest]; found {
```

Comment:
This wouldn’t work as written, `tocDigests[]` is set indexed by the (compressed) blob digest. I suspect the `uncompressedDigest` parameter name is incorrect.

(The whole thing needs to index by layer index within image, anyway… Or maybe use the `tocDigest` value, if the caller already knows it.)

Reply:
The code was replaced, but the misleading parameter name remains.

Reply:
Resolved in #2288.